docs/enhancements/api-for-skipping-validations.md

```diff
@@ -0,0 +1,139 @@
+---
+title: api-for-skipping-validations
+authors:
+  - "@pmaidmen"
+creation-date: 2023-01-08
+last-updated: 2023-01-08
+---
+
+# Allow validations to be skipped
+
+# Summary
+
+We need new api that will allow us to skip cluster/host validations.
+This api should have it's own feature flag to allow us to disable it.
+
+It should be possible to use this API in an unsupported way, it is currently undecided whether or not this will be open or not.
+An event should be emitted whenever the API is used during an installation, this can be used to drive signatures etc.
+# Motivation
+
+Some customer and partners have very specific HW that doesn't pass our validations and we want to allow them to proceed with installation.
+Sometimes we have bugs in our validations that block people from installing and we don't want our partners left in a position where they can't install.
+
+### Goals
+
+- Allow validations to be bypassed so that installation may proceed.
+
+### Non-Goals
+
+- We should not add uneccesary noise to triage tickets with this, it is therefore quite important to add some signatures to indicate that these validations have been bypassed.
+
+## Proposal
+
+I think the API should be kept reasonably simple.
+
+### List existing validations.
+
+GET api/v2/skippedValidations
+Will return a JSON array of all the validations that are currently being skipped, format similar to
+
+```
+{
+  "skipped-host-validations": [
+    "no-ip-collisions-in-network",
+    "sufficient-network-latency-requirement-for-role'"
+  ],
+  "skipped-cluster-validations": [
+    "network-prefix-valid",
+    "api-vips-defined"
+  ]
+}
+```
+
+### Add some skipped validations.
+
+POST api/v2/skippedValidations
```

Comment:
I guess it should be under cluster resource URL?
In this case, would we want to only have this skipping mechanism under the cluster resource? The problem with host validations is that we would prefer to add a skip for all hosts in a cluster, instead of adding it for each host specifically

Reply:
Agreed actually, but ideally we should be able to skip any validation, cluster or host, via this mechanism.
If you specify a host validation to skip, it should be skipped for all hosts.
Is this reasonable?

Reply:
Yes it's also what I would prefer
So the list includes both cluster and host validations, that are skipped for the scope of the whole cluster object. Correct?

Reply:
That's correct.